src/Microsoft.ML.AutoML/AutoMLExperiment/IPerformanceMonitor.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+using System.Timers;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.AutoML
+{
+    public interface IPerformanceMonitor : IDisposable
```

Comment:
Could you please add some comments to the exposed APIs?

Reply:
Actually maybe we should make it internal? I don't expect user to provide their own implementation of that interface.

Reply:
agree.